### PR TITLE
removes api.LegacyUser

### LINKED ISSUE #2260: Remove apiv1.LegacyUser type
**Description**
After removing legacy endpoints (`v3`) `LegacyUser` type is redundant and can be replaced with `apiv1.User` type. 

There is only one endpoint `api/v1/dc` that make use of it though. There is a special case in the handler that checks if the authenticated user is an admin (https://github.com/kubermatic/kubermatic/blob/master/api/pkg/handler/datacenter.go#L29) not sure if this is something that still applies, @mrIncompetent ?

**Acceptance criteria**
 - `LegacyUser` is removed and replaced with `apiv1.User`

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: simply removes unused `api.LegacyUser` type.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2260 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: api/pkg/handler/datacenter.go [hunk 1]
```diff
@@ -26,11 +24,6 @@ func datacentersEndpoint(dcs map[string]provider.DatacenterMeta) endpoint.Endpoi
 		for _, dcName := range keys {
 			dc := dcs[dcName]
 
-			if dc.Private && !IsAdmin(user) {
```
Comment (p0lyn0mial) [inline api/pkg/handler/datacenter.go:29]: This feature was used only internally and it's safe to remove it. Please read the following issue (especially comments)
https://app.zenhub.com/workspaces/all-kubermatic--58863654794153994b62af76/issues/kubermatic/kubermatic/2260

File: api/pkg/handler/datacenter_test.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ func TestDatacentersEndpoint(t *testing.T) {
 		t.Fatalf("Expected route to return code 200, got %d: %s", res.Code, res.Body.String())
 	}
 
-	compareWithResult(t, res, `[{"metadata":{"name":"regular-do1","resourceVersion":"1"},"spec":{"seed":"us-central1","country":"NL","location":"Amsterdam","provider":"digitalocean","digitalocean":{"region":"ams2"}}},{"metadata":{"name":"us-central1","resourceVersion":"1"},"spec":{"seed":"","country":"US","location":"us-central","provider":"digitalocean","digitalocean":{"region":"ams2"}},"seed":true}]`)
+	compareWithResult(t, res, `[{"metadata":{"name":"private-do1","resourceVersion":"1"},"spec":{"seed":"us-central1","country":"NL","location":"US ","provider":"digitalocean","digitalocean":{"region":"ams2"}}},{"metadata":{"name":"regular-do1","resourceVersion":"1"},"spec":{"seed":"us-central1","country":"NL","location":"Amsterdam","provider":"digitalocean","digitalocean":{"region":"ams2"}}},{"metadata":{"name":"us-central1","resourceVersion":"1"},"spec":{"seed":"","country":"US","location":"us-central","provider":"digitalocean","digitalocean":{"region":"ams2"}},"seed":true}]`)
```
Comment (p0lyn0mial) [inline api/pkg/handler/datacenter_test.go:27]: the output is different because i removed checking if the datacenter is private.

**Comment by zreigz** [comment]
The code looks good and also I've tested it locally and works fine. Some e2e tests have failed. Please take a look for it

**Review by zreigz — APPROVED** [review]
lgtm

**Comment by zreigz** [comment]
lgtm

**Pull request merged**